### PR TITLE
Upgrade emotion@10.0.2

### PR DESCRIPTION
**Pull request by lifeiscontent**
Issue:

## What I did

upgraded all packages to use latest version of emotion

## How to test

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by lifeiscontent** [comment]
I couldn't quite get all the tests to pass, I'm really unsure of the details of some of the tests failing, if someone can help me along that'd be great!

**Comment by igor-dv** [comment]
#4525 related

File: docs/src/pages/addons/writing-addons/index.md [hunk 1]
```diff
@@ -238,7 +238,7 @@ storiesOf('Button', module)
 
 We use [emotion](https://emotion.sh) for styling, AND we provide a theme which can be set by the user!
 
-We highly recommend you also use emotion to style your components for storybook, but it's not a requirement. You can use inline styles or another css-in-js lib. You can receive the theme as a prop by using the `withCSSContext` hoc from emotion. [Read more about theming](/configurations/theming).
+We highly recommend you also use emotion to style your components for storybook, but it's not a requirement. You can use inline styles or another css-in-js lib. You can receive the theme as a prop by using the `withTheme` hoc from emotion. [Read more about theming](/configurations/theming).
```
Comment (igor-dv) [inline docs/src/pages/addons/writing-addons/index.md]: I have to say, this looks like a breaking change...

Comment (igor-dv) [reply docs/src/pages/addons/writing-addons/index.md]: @ndelangen I think we need to have some abstraction of the ThemeProvider at our side

**Comment by lifeiscontent** [comment]
@igor-dv anything I can do on my end here or should I just wait till you guys solve some internal issues?

**Comment by igor-dv** [comment]
Let's see what @ndelangen thinks about https://github.com/storybooks/storybook/pull/4925#discussion_r238964042

**Comment by meebix** [comment]
👍 for this. Seeing some strange issues with addons (themed CSS related things) as my project is using Emotion 10, but seems Storybook (and addons) are not.

**Comment by lifeiscontent** [comment]
Cc @igor-dv think you could ping @ndelangen? Would love this to get merged soon

**Comment by igor-dv** [comment]
Yesm will DM him.

**Comment by ndelangen** [comment]
Hey @lifeiscontent thank you for this PR.

I want to merge this, but it does have something of a breaking change in it.
I fear releasing it quickly in a minor will not go down well.

The next major release is scheduled around the 15th of January.

What do you think we should do?

**Comment by ndelangen** [comment]
We can merge this into the overhaul ui branch today, but that's not usable by users yet.

**Comment by lifeiscontent** [comment]
@ndelangen I'm okay with waiting for it, I've currently just disabled all the plugins that break emotion 10. so no knobs for me at the moment, but storybook itself still works.

I actually haven't tested if the code **actually** fixes the issues I'm seeing, is there a way to easily test this code locally within my own project to confirm?

**Comment by ndelangen** [comment]
I merged this (without issue) into #4086 see 2fae294e11e57fd850f9191668279c71c175dcd7

**Comment by igor-dv** [comment]
@ndelangen let's maybe merge this to v5 branch as well?

**Pull request closed without merging**